grails-app/migrations/0.9.x/changelog-2024-07-18-1200-alter-table-location-type-make-location-type-code-non-null.groovy

```diff
@@ -0,0 +1,14 @@
+databaseChangeLog = {
+
+    changeSet(author: "ewaterman", id: "1721321391197-94", objectQuotingStrategy: "LEGACY") {
+        preConditions(onError: "HALT", onFail: "MARK_RAN", onSqlOutput: "IGNORE") {
+            and {
+                columnExists(columnName: "location_type_code", tableName: "location_type")
+
+                sqlCheck("""SELECT COUNT(*) FROM location_type WHERE location_type_code IS NULL""", expectedResult: "0")
```

Comment:
What will happen when there are already location types with code null? Shouldn't it be also fixed?

Reply:
yeah I was thinking about whether or not we want to update all rows that have a code null to have some default code. There aren't any locations like this on obnav, obnavstage, or obdev3 so this will run fine there, but it might still be best to provide a default just in case. The reason I didn't was because I didn't want to silently change behaviour somewhere, but since we're considering location types without a code to be a bug, maybe it's fine. It just felt a little arbitrary picking one at random. Thoughts?

Reply:
Just to update this thread, I changed the script to update any existing location types that don't have a code to use DEPOT. Again, this shouldn't affect any real data (there's no cases of this in prod) but doing it this way makes the script run without errors on any dev environments that happen to have bad data in them